src/System.Memory/tests/Memory/Retain.cs

```diff
@@ -137,5 +137,20 @@ public static void OwnedMemoryRetainWithPinningAndSlice()
             }
             handle.Dispose();
         }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public static void DefaultMemoryRetain(bool pin)
+        {
+            Memory<int> memory = default;
+            MemoryHandle handle = memory.Retain(pin: pin);
+            Assert.False(handle.HasPointer);
```

Comment:
I find this one to be strange. I think the contract is that if you pass true to pin, you will get a pointer. Of course, the pointer would be null in case of default Memory.

But maybe this is not the cotract we want? @stephentoub? @jkotas?

Reply:
> I find this one to be strange. I think the contract is that if you pass true to pin, you will get a pointer. 

To me the strangeness is just the name of the property.  It's really "HasNonNullPointer".